// react-native-paystack-webview/production/lib/utils/helper.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateMetaDataString = exports.paystackHtmlContent = exports.dynamicSplitObjectIsValid = exports.buildKeyValueString = exports.getChannels = exports.getAmountValueInKobo = exports.toAmountInKobo = exports.toString = exports.toNumber = exports.isNegative = exports.isValidDecimalMonetaryValue = exports.isValidStringAmount = void 0;
const validator_1 = __importDefault(require("validator"));
const { isDecimal, isFloat, isInt, toFloat, toInt } = validator_1.default;
function isNumber(value) {
    return typeof value === 'number';
}
function isString(value) {
    return typeof value === 'string';
}
function isValidStringAmount(stringAmount) {
    if (isString(stringAmount) && stringAmount?.endsWith('.')) {
        return false;
    }
    return isDecimal(stringAmount);
}
exports.isValidStringAmount = isValidStringAmount;
function isValidDecimalMonetaryValue(amountValue) {
    if (!isNumber(amountValue) && !isString(amountValue)) {
        return false;
    }
    return isNumber(amountValue) || isValidStringAmount(amountValue);
}
exports.isValidDecimalMonetaryValue = isValidDecimalMonetaryValue;
function isNegative(amountValue) {
    if (typeof amountValue === 'string') {
        return amountValue.startsWith('-');
    }
    return amountValue < 0;
}
exports.isNegative = isNegative;
function toNumber(string) {
    if (isFloat(string)) {
        return toFloat(string);
    }
    if (isInt(string)) {
        return toInt(string);
    }
    return +string;
}
exports.toNumber = toNumber;
function toString(amountValue) {
    return isNumber(amountValue) ? amountValue.toString() : amountValue;
}
exports.toString = toString;
function toAmountInKobo(amountValue) {
    if (typeof amountValue === 'string') {
        return toNumber(amountValue) * 100;
    }
    return amountValue * 100;
}
exports.toAmountInKobo = toAmountInKobo;
const getAmountValueInKobo = (amount) => {
    if (isValidDecimalMonetaryValue(amount)) {
        return toAmountInKobo(amount);
    }
    return 0;
};
exports.getAmountValueInKobo = getAmountValueInKobo;
const getChannels = (channelsArrary) => {
    if (channelsArrary?.length > 0) {
        const channelsString = JSON.stringify(channelsArrary);
        return `channels: ${channelsString},`;
    }
    return '';
};
exports.getChannels = getChannels;
const buildKeyValueString = (key, value) => value ? `${key}: '${value}',` : '';
exports.buildKeyValueString = buildKeyValueString;
const dynamicSplitObjectIsValid = (split) => {
    return (split !== null &&
        typeof split === "object" &&
        split.type &&
        split.bearer_type &&
        Array.isArray(split.subaccounts));
};
exports.dynamicSplitObjectIsValid = dynamicSplitObjectIsValid;
const paystackHtmlContent = (Params) => `
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Paystack</title>
</head>
<body onload="payWithPaystack()" style="background-color:#fff;height:100vh">
  <script src="https://js.paystack.co/v2/inline.js"></script>
  <script>
    function payWithPaystack() {
      var paystack = new PaystackPop();
      paystack.newTransaction({
        ${Params}
        onSuccess: function(response) {
          window.ReactNativeWebView.postMessage(JSON.stringify({
            event: 'successful',
            transactionRef: response,
          }));
        },
        onCancel: function() {
          window.ReactNativeWebView.postMessage(JSON.stringify({
            event: 'cancelled',
          }));
        },
      });
    }
  </script>
</body>
</html>
`;
exports.paystackHtmlContent = paystackHtmlContent;
const generateMetaDataString = (firstName, lastName, billingName, metadata) => {
    return metadata
        ? `metadata: ${metadata},`
        : `metadata: { custom_fields: [{ display_name:  '${firstName + " " + lastName}', variable_name:  '${billingName}', value:'' }]},`;
};
exports.generateMetaDataString = generateMetaDataString;
